### ISSUE TITLE
sometimes the post param dismiss

### ISSUE THREAD

**Issue by lzhAndroid** [square/retrofit#2453]
In retrofit 2.2.0 with kotlin ,sometimes the post param will became 0. 
this is my code:
```
private fun callVideo(context: Context, chatId: Long) {  
            if (chatId == 0L) {  
                ToastUtils.show(context, "something error")  
                RTCControl.instance?.leaveChannel(context)  
            } else {  
                Api.getVideo()  
                        .sendVideoChat(chatId)  
                        .observeOn(AndroidSchedulers.mainThread())  
                        .subscribe({  
                            if (it.ok == BaseEntity.OK) {  
                                startAct(context, chatId, it.call_video_info, true)  
                            } else {  
                                ToastUtils.show(context, it.msg)  
                            }  
                        })  
            }  
        }    
```

```

    @FormUrlEncoded
    @POST("call/send_video_chat")
    Observable<VideoChatBack> sendVideoChat(@Field("receiver_uid") long receiver_uid);
```

the receiver_uid somethings became 0

**Comment by JakeWharton** [comment]
Can you write a test case that demonstrates that this is Retrofit's fault? I have a hard time seeing how we would change the value of a parameter. Are you sure that you're not calling the method from elsewhere with a value of zero?

**Comment by bmarchon** [comment]
I had a similar issue with an @GET interface, also with kotlin. It would somehow lose the values passed as parameters and change them into 0, no multi threading problem for sure. I could not reproduce it through instrumentation test or unit test.

So I switched the interface to java (don't ask me why, last resort solution) and then the application would crash with something similar to these:
- https://github.com/square/retrofit/issues/1506
- https://github.com/square/retrofit/issues/1486

So I disabled instant run and it would just work fine. I reverted my java interface and it worked with kotlin and instant run disabled. 

I would recommend to try disabling instant run, hopefully it will solve your issue. If not maybe try to change gradle plugin version.
